src/main/java/os/failsafe/executor/FailsafeExecutor.java

```diff
@@ -612,11 +615,41 @@ private void notifyPersisting(Task task, String taskId) {
         listeners.forEach(listener -> listener.persisting(name, taskId, parameter));
     }
 
-    private void validateDatabaseTableStructure(DataSource dataSource) throws SQLException {
+    private void validateDatabase(DataSource dataSource) throws SQLException {
+        List<Task> tasks = new ArrayList<>();
+
+        // validate table structure (with persist)
         try (Connection connection = dataSource.getConnection();
              Transaction transaction = new Transaction(connection)) { // no commit of trx
-            Task testTask = new Task(UUID.randomUUID().toString(), "validateDatabaseTableTaskName", null, systemClock.now());
-            taskRepository.add(connection, testTask);
+            tasks.add(new Task(UUID.randomUUID().toString(), "validateDatabaseTableTaskName", null, systemClock.now()));
+            tasks.add(new Task(UUID.randomUUID().toString(), "validateDatabaseTableTaskName", null, systemClock.now()));
+            for (Task task : tasks) {
+                if (taskRepository.add(connection, task) == null) {
+                    throw new IllegalStateException("Validation of database failed! Unable to persist tasks!");
```

Comment:
I would give a reference to the mariadb issue that explains the situation, within the ExceptionMessage.